content/en/docs/components/pipelines/sdk/build-pipeline.md

```diff
@@ -0,0 +1,465 @@
+function][python-function-component]. You can also [reuse prebuilt components
+in your pipeline][prebuilt-components]. 
+
+### Understanding the pipeline graph
+
+Each step in your pipeline's workflow is an instance of a component. When
+you define your pipeline, you specify the source of each step's inputs. Step
+inputs can be set from the pipeline's input arguments, constants, or step
+inputs can depend on the outputs of other steps in this pipeline. Kubeflow
+Pipelines uses these dependencies to define your pipeline's workflow as
+a graph.
+
+For example, consider a pipeline with the following steps: ingest data,
+generate statistics, preprocess data, and train a model. The following
+describes the data dependencies between each step.
+
+*   **Ingest data**: This step loads data from an external source which is
+    specified using a pipeline argument, and it outputs a dataset. Since
+    this step does not depend on the output of any other steps, this step
+    can run first.
+*   **Generate statistics**: This step uses the ingested dataset to generate
+    and output a set of statistics. Since this step depends on the dataset
+    produced by the ingest data step, it must run after the ingest data step.
+*   **Preprocess data**: This step preprocesses the ingested dataset and
+    transforms the data into a preprocessed dataset. Since this step depends
+    on the dataset produced by the ingest data step, it must run after the
+    ingest data step.
+*   **Train a model**: This step trains a model using the preprocessed dataset,
+    the generated statistics, and pipeline parameters, such as the learning
+    rate. Since this step depends on the preprocessed data and the generated
+    statistics, it must run after both the preprocess data and generate
+    statistics steps are complete.
+
+Since the generate statistics and preprocess data steps both depend on the
+ingested data, the generate statistics and preprocess data steps can run in
+parallel. All other steps are executed once their data dependencies are
+available.
+
+## Designing your pipeline
+
+When designing your pipeline, think about how to split your ML workflow into
+pipeline components. The process of splitting an ML workflow into pipeline
+components is similar to the process of splitting a monolithic script into
+testable functions. The following rules can help you define the components
+that you need to build your pipeline.
+
+*   Components should have a single responsibility. Having a single
+    responsibility makes it easier to test and reuse a component. For example,
+    if you have a component that loads data you can reuse that for similar
+    tasks that load data. If you have a component that loads and transforms
+    a dataset, the component can be less useful since you can use it only when
+    you need to load and transform that dataset. 
+
+*   Reuse components when possible. Kubeflow Pipelines provides [components for
```

Comment:
Just wanted to double check that "Kubeflow Pipelines" is "it" and not "they" (hence, "provides" and not "provide").

Reply:
Yes, in this context "Kubeflow Pipelines" is singular.